seed_data/management/commands/alter_data.py

```diff
@@ -178,26 +185,43 @@ def clear_future_runs(user=None, course=None):  # pylint: disable=unused-argumen
 @accepts_or_calculates_now
 def add_past_failed_run(user=None, course=None, now=None, grade=DEFAULT_FAILED_GRADE, add_if_exists=False):
     """Adds a past failed course run for a given user and course"""
+    chosen_past_run = get_past_course_run(user, course, now, add_if_exists)
+
+    # Add enrollment and failed grade for course run
+    CachedEnrollmentHandler(user).set_or_create(course_run=chosen_past_run)
+    CachedCurrentGradeHandler(user).set_or_create(course_run=chosen_past_run, grade=grade)
+    return chosen_past_run
+
+
+def get_past_course_run(user=None, course=None, now=None, add_if_exists=False):
```

Comment:
one other thing i just noticed: this function should be checking for both current grade and certificates. here's a snippet that includes that and some minor cleanup:

```python 
def get_past_course_run(user=None, course=None, now=None, add_if_exists=False):
    """Loop through past course runs and find one without grade data"""
    past_runs = CourseRun.objects.filter(
        course=course,
        end_date__lt=now,
    ).exclude(end_date=None).order_by('-end_date').all()
    for past_run in past_runs:
        if not CachedCurrentGradeHandler(user).exists(past_run) and \
                not CachedCertificateHandler(user).exists(past_run):
            return past_run
        elif not add_if_exists:
            raise Exception("Past failed/passed course run already exists (id: {}, title: {})".format(
                past_run.id,
                past_run.title
            ))
    else:
        raise Exception("Can't find past run w/o grade data")
```